### DISCUSSION TITLE
setting baseBranch still takes renovate.json from default branch instead of new baseBranch

### DISCUSSION THREAD

**Discussion by ekhaydarov** [renovatebot/renovate#11761] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### Please select which platform you are using if self-hosting.

github.com

### If you're self-hosting Renovate, tell us what version of Renovate you run.

27.6.2

### Describe the bug

when setting `baseBranch` it still takes the renovate.json from the default branch of the repository and not directed at the `baseBranch` set in the config

### Relevant debug logs

pretty evident when nothing was changing when i kept pushing changes to `renovate.json5` file in the new `baseBranch` 

```
DEBUG: Config file exists (repository=farfetch-internal/data-infrastructure-core)
"fileName": "renovate.json"
```

### Have you created a minimal reproduction repository?

No reproduction repository

**Comment by viceice** [comment, marked as answer]
This is by design, as we don't know baseBranches before we fetch the config from default branch

**Reply by viceice** [reply]
#2496

**Reply by ekhaydarov** [reply]
so we are meant to commit to master branch until we get it right?

**Reply by viceice** [reply]
yes, you can also fork your repo and test against  it ;)

**Reply by ekhaydarov** [reply]
some private orgs have forking disabled. 

In any case the issue seems to state that we should be able to use `useBaseBranchConfig`?

**Reply by viceice** [reply]
nope, `useBaseBranchConfig` is not yet implemented, thats why the issue is still open
